snowfall/decoding/rescore.py

```diff
@@ -191,10 +202,31 @@ def rescore(lats: k2.Fsa,
         tot_scores_2nd_num = reorded_lats.get_tot_scores(
             use_double_scores=True, log_semiring=True)
 
+        for k in [0, 1, 2, 30, 40, 50]:
```

Comment:
Output log of this `for` loop is attached below:

[log-decode-second-2021-03-19-21-25-09.txt](https://github.com/k2-fsa/snowfall/files/6171426/log-decode-second-2021-03-19-21-25-09.txt)


A part of them are listed as follows:
```
2021-03-19 21:25:30,611 INFO [rescore.py:209] 
path: 0
tot_scores: -inf
log_probs:[ [ -86.4828 -2.01945 -11.1086 -133.909 -8.51447 -4.66334  .....  -6.37545 -2.73227 -45.2354 -5.78217 ] ]

2021-03-19 21:25:30,612 INFO [rescore.py:209] 
path: 1
tot_scores: -inf
log_probs:[ [ -87.2116 -1.99367 -8.34123 -134.502 -8.31313 ....   -2.12719 -6.3754 -2.41306 -27.2158 -5.17843 ] ]

2021-03-19 21:25:30,612 INFO [rescore.py:209] 
path: 2
tot_scores: -inf
log_probs:[ [ -87.5116 -2.04247 -8.2759 -134.616 -8.62277 ..... -5.99715 -2.10805 -5.69009 -1.49818 -44.5129 -5.55875 ] ]

2021-03-19 21:25:30,613 INFO [rescore.py:209] 
path: 30
tot_scores: -358.8836602834303
log_probs:[ [ -166.046 -3.19678 -5.97864 -219.513 -6.48385  ...  -1.07081 -4.60576 -2.2935 -0.218605 -5.64143 ] ]

2021-03-19 21:25:30,613 INFO [rescore.py:209] 
path: 40
tot_scores: -401.60242305657346
log_probs:[ [ -112.147 -2.33504 -3824.3 -177.408 -9.14423 -3678.67 -6.93939 .... -2.56483 -1.64079 -4.76889 ] ]

2021-03-19 21:25:30,613 INFO [rescore.py:209] 
path: 50
tot_scores: -398.0919782588134
log_probs:[ [ -138.47 -3.37409 -5.20867 -219.942 -7.50989 -1.4931  ....   -5.48529 -2.04127 -2.47733 -4.61414 ] ]
```

I am not sure whether these log-probs look reasonable.

Reply:
The numbers I was expecting would be quite close to zero, and even closer at odd positions (or maybe even.. i.e. where there are epsilons).  I.e. I mean the posterior of the "reference phone" at each position (it's not really the reference, it's the sequence we use for alignment0.

Reply:
I was expecting you'd get it by indexing second_pass_dense_fsa_vec with some kind of tensor that's related to the reference phones.  Or perhaps you could just take the sum over a particular axis, of (second_pass_dense_fsa_vec * phone_one_hot_input).